element_optogenetics/optogenetics.py

```diff
@@ -1,159 +1,234 @@
+                            does not yet exist.
+        create_tables (bool): when True (default), create schema tables in the database
+                             if they do not yet exist.
+        linking_module (str): a module (or name) containing the required dependencies.
+
+    Dependencies:
+        Upstream tables:
+            Device: Referenced by Protocol. Device to perform procedure
+            Session: Parent to Session. Typically identifying a recording session
+            Trial: Parent to OptoTrial. Optionally from Element-Event
+            Event: Parent to OptoTrial. Optionally from Element-Event
+            SkullReference: Referenced by Session.SkullReference.
+                Specifying the skull reference, such as bregma or lambda
     """
-    activate(schema_name, create_schema=True, create_tables=True)
-        :param schema_name: schema name on the database server to activate the `optogenetics` element
-        :param create_schema: when True (default), create schema in the database if it does not yet exist.
-        :param create_tables: when True (default), create tables in the database if they do not yet exist.
-        :param linking_module: a module name or a module containing the
-         required dependencies to activate the `subject` element:
-             Upstream tables:
-                + Device: Reference table for OptoProtocol, device to perform optogenetics experiments
-                + Session: Parent table to OptoSession, typically identifying a recording session
-                + SessionTrial: Parent table to OptoTrial, passive trial or behavioral trial
-                + BrainRegion: Reference table for OptoSession.BrainRegion, specifying the skull reference, such as bregma or lambda
-                + SkullReference: Reference table for OptoSession.BrainLocation, specifying the brain region
 
+    if isinstance(linking_module, str):
+        linking_module = importlib.import_module(linking_module)
+    assert inspect.ismodule(
+        linking_module
+    ), "The argument 'dependency' must be a module's name or a module"
 
-    """
-    schema.activate(schema_name, create_schema=create_schema, create_tables=create_tables)
+    global _linking_module
+    _linking_module = linking_module
+
+    # activate
+
+    schema.activate(
+        schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        add_objects=_linking_module.__dict__,
+    )
 
 
 @schema
-class OptoWaveformType(dj.Lookup):
+class WaveformType(dj.Lookup):
+    """
+    Attributes:
+        waveform_type ( varchar(32) ): Waveform type (e.g., square, sine)
+    """
+
     definition = """
-    opto_waveform_type:    varchar(32)
+    waveform_type:    varchar(32)
     """
-    contents = zip(['Square', 'Ramp', 'Sine'])
+    contents = zip(["Square", "Ramp", "Sine"])
 
 
 @schema
-class OptoWaveform(dj.Lookup):
+class Waveform(dj.Lookup):
```

Comment:
In general, I'm slightly in favor of sticking with `OptoWaveform ` to prevent possible confusion when this gets paired with ephys.

Reply:
Makes sense. I'll revert to include the Opto specifier. Thanks @JaerongA